File: jbake-core/src/main/java/org/jbake/app/Utensils.java
package org.jbake.app;

import org.jbake.app.configuration.JBakeConfiguration;

/**
 * A helper class to wrap all the utensils that are needed to bake.
 */
public class Utensils {
    private JBakeConfiguration configuration;
    private ContentStore contentStore;
    private Crawler crawler;
    private Renderer renderer;
    private Asset asset;

    public JBakeConfiguration getConfiguration() {
        return configuration;
    }

    public void setConfiguration(JBakeConfiguration configuration) {
        this.configuration = configuration;
    }

    public ContentStore getContentStore() {
        return contentStore;
    }

    public void setContentStore(ContentStore contentStore) {
        this.contentStore = contentStore;
    }

    public Crawler getCrawler() {
        return crawler;
    }

    public void setCrawler(Crawler crawler) {
        this.crawler = crawler;
    }

    public Renderer getRenderer() {
        return renderer;
    }

    public void setRenderer(Renderer renderer) {
        this.renderer = renderer;
    }

    public Asset getAsset() {
        return asset;
    }

    public void setAsset(Asset asset) {
        this.asset = asset;
    }
}

File: jbake-core/src/main/java/org/jbake/app/Utensils.kt
package org.jbake.app

import org.jbake.app.configuration.JBakeConfiguration
import org.jbake.app.configuration.JBakeConfigurationInspector


/** A helper class to wrap all the utensils that are needed to bake. */
class Utensils(
    val configuration: JBakeConfiguration,
    val contentStore: ContentStore,
    val crawler: Crawler,
    val renderer: Renderer,
    val asset: Asset
)



/**
 * A factory to create a [Utensils] object.
 */
object UtensilsFactory {
    /**
     * Create default [Utensils] by a given [JBakeConfiguration]
     * @return a default [Utensils] instance
     */
    fun createDefaultUtensils(config: JBakeConfiguration): Utensils {
        val inspector = JBakeConfigurationInspector(config)
        inspector.inspect()

        val contentStore = DbUtils.createDataStore(config)
        return Utensils(
            configuration = config,
            contentStore = contentStore,
            crawler = Crawler(contentStore, config),
            renderer = Renderer(contentStore, config),
            asset = Asset(config)
        )
    }
}
